analyzer/src/main/kotlin/PackageManager.kt

```diff
@@ -83,14 +83,23 @@ abstract class PackageManager {
                     val filesInDir = dir.toFile().listFiles()
 
                     packageManagers.forEach { manager ->
-                        val matches = manager.matchersForDefinitionFiles.mapNotNull { glob ->
-                            filesInDir.find { file ->
-                                glob.matches(file.toPath())
+                        // Create a list of lists of matching files per glob.
+                        val matchesPerGlob = manager.matchersForDefinitionFiles.mapNotNull { glob ->
+                            // Create a list of files in the current directory that match the current glob.
+                            val filesMatchingGlob = filesInDir.filter { file ->
+                                file != null && glob.matches(file.toPath())
```

Comment:
Can file be null? Just ask because this was not checked before and I never saw an NPE here.

Reply:
Well, it shouldn't, but it's a nullable Java type, which is what I wanted to emphasize here.